framework/web/UrlManager.php

```diff
@@ -315,7 +315,12 @@ public function createUrl($params)
         $baseUrl = $this->showScriptName || !$this->enablePrettyUrl ? $this->getScriptUrl() : $this->getBaseUrl();
 
         if ($this->enablePrettyUrl) {
-            $cacheKey = $route . '?' . implode('&', array_keys($params));
+            $cacheKey = $route . '?';
+            foreach($params as $key => $value) {
+                if ($value !== null) {
+                    $cacheKey .= $key . '&';
```

Comment:
This line causes redundant `&` at the end of the line, but this algorithm is the fastest and it does not matter.